### DISCUSSION TITLE
Can't figure out looking ahead

### DISCUSSION THREAD

**Discussion by IsotoxalDev** [tree-sitter/tree-sitter#2103] [Q&A]
I am creating a parser based on PEST. in that it was created like:
![image](https://user-images.githubusercontent.com/62714538/221372678-9104abb3-c985-4a0e-8231-72e588daa9d2.png)
I tried a number of ways to do it in Tree sitter by doing precedence and choice. Also looked into external, but i want rules I have written in the grammer.js file to check or rewrite entirely in scanner.cc which would make it work but would lead me to spent another 8 hours(it took that to just do the same in grammer.js)

**Comment by ahelwer** [comment]
What is some example input you are trying to parse, what is your desired syntax tree, and what is the syntax tree you are getting instead?

**Reply by IsotoxalDev** [reply]
```
Test {
   plaint_text till a comment // A comment
  plain_text till multiline comment /* A
Multiline Comment */ plain text till another element start Elem {A new element}
plain text till end of svope
}
```
Expected tree
```
element
  plain_text
  comment
  plain_text
  comment
  plain_text
  elememnt
  plain_text
```

**Reply by ahelwer** [reply]
Okay, what have you tried and why does it not work?

**Reply by IsotoxalDev** [reply]
```
        plain_text: $ => /(.|\n)/,

        element_start: $ => seq(
            $.namespaced_identifier,
            optional($.element_id),
            repeat($.element_class),
            optional($.element_attributes),
            $.curly_bracket_open
        ),

        structure_child: $ => choice(
            $.component_element,
            $.structure_element,
            $.style_element,
            choice(
                prec(3, choice(
                    $.element_start,
                    $.comment,
                    $.curly_bracket_close,
                )),
                prec(2, $.plain_text),
            )
        ),

``` 
This is what i tried and it gives the whole source cod as an error
